reader.go

```diff
@@ -30,6 +31,8 @@ type Reader struct {
 	dbfNumRecords   int32
 	dbfHeaderLength int16
 	dbfRecordLength int16
+
+	memoryData *memoryShapeData
```

Comment:
I'm not happy with adding this member variable to the struct.

Do you need to have random access? If not, have you considered https://godoc.org/github.com/jonas-p/go-shp#SequentialReaderFromExt for opening the byte slices?